src/domain/swaps/contracts/decoders/gp-v2-decoder.helper.ts

```diff
@@ -660,22 +657,13 @@ export class GPv2Decoder extends AbiDecoder<typeof GPv2Abi> {
   public getOrderUidFromSetPreSignature(
     data: `0x${string}`,
   ): `0x${string}` | null {
-    if (!this.helpers.isSetPreSignature(data)) {
-      return null;
-    }
-
-    try {
-      const decoded = this.decodeFunctionData({ data });
-
-      if (decoded.functionName !== 'setPreSignature') {
-        throw new Error('Data is not of setPreSignature');
-      }
+    const decoded = this.decodeFunctionData.setPreSignature(data);
 
-      return decoded.args[0];
-    } catch (e) {
-      this.loggingService.debug(e);
+    if (!decoded) {
       return null;
```

Comment:
Nit: Wdyt about logging a warning in this situation?

Reply:
This shouldn't throw because of the initial `isSetPreSignature` so logging isn't needed.